### PR TITLE
Added option for the logging in Fixes #709

### LINKED ISSUE #709: Server log / stack trace in e2e tests
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

When I run e2e tests against an endpoint and the app throws a 500 Internal Server Error, I's very difficult to debug where the issue arises since the stack trace only includes references to supertest source. 

```
Error: expected 200 "OK", got 500 "Internal Server Error"
          at Test.Object.<anonymous>.Test._assertStatus (/Users/bostrom/Projects/Grano/marketing-api/node_modules/supertest/lib/test.js:268:12)
          at Test.Object.<anonymous>.Test._assertFunction (/Users/bostrom/Projects/Grano/marketing-api/node_modules/supertest/lib/test.js:283:11)
          at Test.Object.<anonymous>.Test.assert (/Users/bostrom/Projects/Grano/marketing-api/node_modules/supertest/lib/test.js:173:18)
          at Server.assert (/Users/bostrom/Projects/Grano/marketing-api/node_modules/supertest/lib/test.js:131:12)
          at Object.onceWrapper (events.js:273:13)
          at Server.emit (events.js:182:13)
          at Server.EventEmitter.emit (domain.js:442:20)
          at emitCloseNT (net.js:1649:8)
          at process._tickCallback (internal/process/next_tick.js:63:19)
```

I don't see any other ways of debugging than throwing a bunch of consoe.logs in the code to try to figure out where the problem lies. 

Also, a cool bonus would be to enable the node debugger/inspector for e2e tests!

## Expected behavior

When I test the endpoint using the actual app, the issue produces a relevant stack trace in the console where the app was started. This is what I would like to see when a test fails due to an application error. 

## Minimal reproduction of the problem with instructions
Create an app with an bug in the application, e.g. referencing a variable that doesn't exist. Run an e2e test against it and start scratching your head.

## What is the motivation / use case for changing the behavior?
Faster debugging and time to production.

## Environment

<pre><code>
Nest version: 5.0.0

 
For Tooling issues:
- Node version: 10.0.1  
- Platform: Mac  

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by httpsOmkar**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[*] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #709 

## Does this PR introduce a breaking change?
```
[ ] Yes
[*] No
```

## Other information

### CODE REVIEW

File: packages/testing/test.ts [hunk 1]
```diff
@@ -9,16 +9,17 @@ import { Logger } from '@nestjs/common/services/logger.service';
...
 import { MetadataScanner } from '@nestjs/core/metadata-scanner';
 import { TestingModuleBuilder } from './testing-module.builder';
+import { TestOptions } from './interfaces/test-options.interface';
 
 export class Test {
   private static readonly metadataScanner = new MetadataScanner();
 
-  public static createTestingModule(metadata: ModuleMetadata) {
-    this.init();
+  public static createTestingModule(metadata: ModuleMetadata, testOptions: TestOptions) {
```
Comment (shekohex) [inline packages/testing/test.ts]: Oh, Could you make that optional ?
Because that will require -after updating- adding that option to all tests.

Comment (httpsOmkar) [reply packages/testing/test.ts]: Sure

**Comment by kamilmysliwiec** [comment]
Hi @httpsOmkar,
Thanks for your effort, great stuff! However, this issue was larger than I expected. Basically, we have been completely ignoring custom logger property. It's fixed in `v5.0.1` which makes this pull request outdated. Even though I can't merge it, thank you for your time 🔥

**Comment by httpsOmkar** [comment]
This is my first of coding not typo PR request to any open source project it seems fun. I will try to make more PR request. you really add some awsome. And you really added great stuff in `v5.0.1`.

**Comment by kamilmysliwiec** [comment]
Thanks @httpsOmkar, looking forward to more PR's!! 👍

**Pull request closed without merging**